### DISCUSSION TITLE
Feedback on initial front end (Feb 7 showcase)

### DISCUSSION THREAD

**Discussion by scottveirs** [orcasound/bioacoustic-dashboard#3] [Ideas]
What does everyone think about this "first draft" of the bioacoustic dashboard from @jonom ? 

[https://preview.menz.dev/soundspace-analytics/dashboard-fe-poc/](https://preview.menz.dev/soundspace-analytics/dashboard-fe-poc/)

Soliciting feedback from beta-testers:
@veirs 
@scottveirs 
And Janie (who Scott will onboard)... ;)

Thanks for the great tour this morning, @ben-hendricks !

**Comment by scottveirs** [comment]
Summary statics view:

<img width="512" alt="Screenshot 2023-02-07 at 12 08 31 PM" src="https://user-images.githubusercontent.com/14044595/217353879-510d3f12-b941-48d4-93df-f3327a25572a.png">

**Comment by scottveirs** [comment]
Detection time series view:
<img width="1043" alt="Screenshot 2023-02-07 at 12 09 10 PM" src="https://user-images.githubusercontent.com/14044595/217354007-280a5917-1bb4-419b-b24d-b6c92c70e7b9.png">

**Reply by scottveirs** [reply]
While I'm not a huge fan of stacked bar charts, I see the utility of aggregating and visualizing all types of detections this way. However, in order to visualize temporal correlations between signals, I would like to see a slight modification to this Rolling Average version of the `Detection time series` view:

<img width="765" alt="Screenshot 2023-02-07 at 12 21 47 PM" src="https://user-images.githubusercontent.com/14044595/217357283-bcb8cbcc-5f6d-4c49-835c-b5ab83c26415.png">

**If possible, could the lines be plotted with their current colors, but without the semi-transparent shading?**

If there's no technical reason why it can't be done, @jonom, I think it may help look for patterns -- especially if/when more signal types are added. As is, I find it a bit difficult to trace how each line goes through the more complicated areas where three shadings overlap. It gets a bit worse as you zoom out:

<img width="774" alt="Screenshot 2023-02-07 at 12 31 22 PM" src="https://user-images.githubusercontent.com/14044595/217358399-7ff9fd9b-07bc-45ec-bb70-15d137a7ee5c.png">

That said, I think the semi-transparent shading works well for the simpler curves in the `Interpolated average` view:

<img width="765" alt="Screenshot 2023-02-07 at 12 30 59 PM" src="https://user-images.githubusercontent.com/14044595/217358445-31dee970-d515-463e-9ce6-14d1382fa759.png">

**Reply by scottveirs** [reply]
@jonom  and @ben-hendricks -- Wondering if this sort of Github discussion of draft plots/views is helpful to you, or if you'd prefer specific issues be filed as beta-testing continues?

If it is useful for general feedback and initial brainstorming of feature requests (later submitted as well-formed issues), then maybe one approach that's even more secure than a shared password approach would be to just screenshot each of the views you demoed last weekend in a new Discussion thread, one by one?

**Comment by scottveirs** [comment]
Acoustic events view:

<img width="1014" alt="Screenshot 2023-02-07 at 12 09 46 PM" src="https://user-images.githubusercontent.com/14044595/217354116-c9df632b-7e28-4e42-b5f6-1ac1366e31b4.png">

**Reply by scottveirs** [reply]
vs the previous version:

<img width="1026" alt="Screenshot 2023-02-07 at 12 10 13 PM" src="https://user-images.githubusercontent.com/14044595/217354217-e7b671f3-3b0a-4615-8dd5-887ffdfedfc2.png">

<img width="1027" alt="Screenshot 2023-02-07 at 12 11 16 PM" src="https://user-images.githubusercontent.com/14044595/217354387-25e64153-65f6-4ff0-a3c8-5e2497075241.png">

**Reply by scottveirs** [reply]
@jonom Just capturing what Grace said during the launch workshop:

As a field scientist interested in the different uses of humpback non-song vocalization (i.e. bubble-net feeding calls vs social calls), she would like some way to visually differentiate between them in the bubble charts. Your proposed solution was to make more dramatic the differences in the fill color of the bubbles.

Other possible solutions:

1. Change the outline color of the bubbles to delineate 2+ different signals within a vocalizing species
2. Tom suggesting changing the shape of the bubble (to which Ben said there are no square bubbles!)
3. Provide some UI or configuration file (without adding too much complexity to the code) that enables simpler or more complex color palettes/preferences?

**Comment by scottveirs** [comment]
Spectral averages (LTSA) view:

<img width="677" alt="Screenshot 2023-02-07 at 12 11 46 PM" src="https://user-images.githubusercontent.com/14044595/217354531-12665ad8-b22e-4a4f-bf8b-42d8735856ea.png">

**Comment by scottveirs** [comment]
Sound level timeline view:

<img width="1015" alt="Screenshot 2023-02-07 at 12 12 27 PM" src="https://user-images.githubusercontent.com/14044595/217354650-d1ac1f47-cc8f-41f3-92c3-8f9463de674f.png">

**Comment by scottveirs** [comment]
Recording consistency view:

<img width="998" alt="Screenshot 2023-02-07 at 12 13 04 PM" src="https://user-images.githubusercontent.com/14044595/217354752-9cf84d96-9a01-48f1-91fd-a2531a9bbf5d.png">

**Comment by ben-hendricks** [comment]
Hi Scott and Jono,

Jono is in the very last (financial) stages of his initial contract by now, so my suggestion is to have Jono’s focus on finalizing functionality, basic documentation as well as creating a version 1 front-end design for the dashboard which we can use to launch it as part of the BC hydrophone network framework. 

Realistically Jono will only implement specific issues (= problems) that are immediate to functionality and UX. But since the dashboard will start its own life as an Open Source product very soon, I am not opposed to initiate a general discussions section anytime for future contributors to pick up and work on (or through a follow-up contract with Jono).

Cheers,
Ben

> On Apr 18, 2023, at 9:42 AM, Scott Veirs ***@***.***> wrote:
> 
> 
> @jonom <https://github.com/jonom> and @ben-hendricks <https://github.com/ben-hendricks> -- Wondering if this sort of Github discussion of draft plots/views is helpful to you, or if you'd prefer specific issues be filed as beta-testing continues?
> 
> If it is useful for general feedback and initial brainstorming of feature requests (later submitted as well-formed issues), then maybe one approach that's even more secure than a shared password approach would be to just screenshot each of the views you demoed last weekend in a new Discussion thread, one by one?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/orcasound/bioacoustic-dashboard/discussions/3#discussioncomment-5651455>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A374DCJPLY52FRUDFDEMGRTXB275RANCNFSM6AAAAAAUUMZYGI>.
> You are receiving this because you were mentioned.
> 


-- 
Benjamin Hendricks, PhD__________________________
*SoundSpace Analytics*
***@***.*** 
***@***.***>
(+1) 250 532 3179

**Comment by ben-hendricks** [comment]
My preference would be Option 3…and I would still veto Option 2.

Another option may be to limit the number of simultaneously shown signal classes to 2. The user can select a class to display and a maximum of one comparison. The colors are hard coded, easily to differentiate and work for the color-blind. In my experience displaying more than 2 sets of data at the same time almost always results in loss of clarity no matter the color-scheme, symbol shape etc.

But as I have written in my previous email, this may be beyond what we can ask of Jono now, and the quickest way to generate user flexibility is a config file (or a clearly visible place in the source code) so end users can configure for themselves.

- Ben

> On Apr 18, 2023, at 9:39 AM, Scott Veirs ***@***.***> wrote:
> 
> 
> @jonom <https://github.com/jonom> Just capturing what Grace said during the launch workshop:
> 
> As a field scientist interested in the different uses of humpback non-song vocalization (i.e. bubble-net feeding calls vs social calls), she would like some way to visually differentiate between them in the bubble charts. Your proposed solution was to make more dramatic the differences in the fill color of the bubbles.
> 
> Other possible solutions:
> 
> Change the outline color of the bubbles to delineate 2+ different signals within a vocalizing species
> Tom suggesting changing the shape of the bubble (to which Ben said there are no square bubbles!)
> Provide some UI or configuration file (without adding too much complexity to the code) that enables simpler or more complex color palettes/preferences?
> —
> Reply to this email directly, view it on GitHub <https://github.com/orcasound/bioacoustic-dashboard/discussions/3#discussioncomment-5651430>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A374DCODGTLRLUDJY3OC7MLXB27T5ANCNFSM6AAAAAAUUMZYGI>.
> You are receiving this because you were mentioned.
> 


-- 
Benjamin Hendricks, PhD__________________________
*SoundSpace Analytics*
***@***.*** 
***@***.***>
(+1) 250 532 3179
